guides/v2.2/magento-functional-testing-framework/release-2/extending.md

```diff
@@ -0,0 +1,377 @@
+        </assertCount>
+    </actionGroup>
+</actionGroups>
+```
+
+### Add an action
+
+__Use case__: The `GetProductCount` action group grabs the count of products.
+Add a new test `VerifyProductCount` that will assert the count of products:
+
+> Action groups with "extends":
+
+```xml
+<actionGroups>
+    <actionGroup name="GetProductCount">
+        <arguments>
+            <argument name="productSelector" type="string"/>
+        </arguments>
+        <grabMultiple selector="{{productSelector}}" stepKey="grabProducts"/>
+    </actionGroup>
+
+    <actionGroup name="VerifyProductCount" extends="GetProductCount">
+        <arguments>
+            <argument name="count" type="string"/>
+        </arguments>
+        <assertCount stepKey="assertCount" after="grabProducts">
+            <expectedResult type="int">{{count}}</expectedResult>
+            <actualResult type="variable">grabProducts</actualResult>
+        </assertCount>
+    </actionGroup>
+</actionGroups>
+```
+
+> Action groups without "extends":
+
+```xml
+<actionGroups>
+    <actionGroup name="GetProductCount">
+        <arguments>
+            <argument name="productSelector" type="string"/>
+        </arguments>
+        <grabMultiple selector="{{productSelector}}" stepKey="grabProducts"/>
+    </actionGroup>
+
+    <actionGroup name="VerifyProductCount">
+        <arguments>
+            <argument name="count" type="string"/>
+            <argument name="productSelector" type="string"/>
+        </arguments>
+        <grabMultiple selector="{{productSelector}}" stepKey="grabProducts"/>
+        <assertCount stepKey="assertCount">
+            <expectedResult type="int">{{count}}</expectedResult>
+            <actualResult type="variable">grabProducts</actualResult>
+        </assertCount>
+    </actionGroup>
+</actionGroups>
+```
+
+## Extending data
+
+Extend data to reuse entities in your module.
+
+### Update a data entry
+
+__Use case__: Create the `DivPanelGreen` entity that is similar to the `DivPanel` entity but of a different color.
+
+> Entities with "extends":
+
+```xml
+<entities >
+    <entity name="DivPanel">
+        <data key="divColor">Red</data>
+        <data key="divSize">80px</data>
+        <data key="divWidth">100%</data>
+    </entity>
+    <entity name="DivPanelGreen" extends="DivPanel">
+        <data key="divColor">Green</data>
+    </entity>
+</entities>
+```
+
+> Entities without "extends":
+
+```xml
+<entities >
+    <entity name="DivPanel">
+        <data key="divColor">Red</data>
+        <data key="divSize">80px</data>
+        <data key="divWidth">100%</data>
+    </entity>
+    <entity name="DivPanelGreen" extends="DivPanel">
+        <data key="divColor">Green</data>
+        <data key="divSize">80px</data>
+        <data key="divWidth">100%</data>
+    </entity>
+</entities>
+```
+
+### Add a data entry
+
+__Use case__: Create the `DivPanelGreen` entity that is similar to the `DivPanel` entity but with a specific panel color.
```

Comment:
Same as above, except for "panel color" instead of "color"

Reply:
declined, with the same reason as above